curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c8ee68f280c0a58a3d439.md

```diff
@@ -0,0 +1,153 @@
+---
+id: 643c8ee68f280c0a58a3d439
+title: Step 29
+challengeType: 0
+dashedName: step-29
+---
+
+# --description--
+
+To print the result on the page, target the HTML element with the `id` of `posts-container`.
+
+# --hints--
+
+You should use a `const` variable named `postsContainer` to target the HTML element with the `id` of `posts-container`.
+
+```js
+assert(code.match(/const\s+postsContainer\s+=\s+document\.getElementById\(['"]posts-container['"]\)/g));
+```
```

Comment:
````suggestion
You should use `const` to declare a `postsContainer` variable.

```js
assert.match(code, /const\s+postsContainer\s+=/);
```

You should use `document.getElementById()` to get the `#posts-container` element.

```js
assert.match(code, /document\.getElementById\(('|"|`)posts-container\1\)/);
```

You should assign the `#posts-container` element to your `postsContainer` variable.

```js
assert(code.match(/const\s+postsContainer\s+=\s+document\.getElementById\(('|"|`)posts-container\1\)/g));
```
````

Prefer granular steps. It helps campers identify where a typo might be, for example.

Reply:
fixed